shapes/main.cpp

```diff
@@ -1,96 +1,85 @@
-#include <iostream>
-#include <vector>
 #include <algorithm>
-#include <string>
+#include <functional>
+#include <iostream>
+#include <map>
 #include <memory>
-#include "Shape.hpp"
+#include <string>
+#include <vector>
+#include "Circle.hpp"
 #include "Rectangle.hpp"
+#include "Shape.hpp"
 #include "Square.hpp"
-#include "Circle.hpp"
 
 using namespace std;
 
 using Collection = vector<shared_ptr<Shape>>;
 
-bool sortByArea(shared_ptr<Shape> first, shared_ptr<Shape> second)
-{
-    if(first == nullptr || second == nullptr)
+auto sortByArea = [](shared_ptr<Shape> first, shared_ptr<Shape> second) {
+    if (first == nullptr || second == nullptr) {
         return false;
+    }
     return (first->getArea() < second->getArea());
-}
+};
 
-bool perimeterBiggerThan20(shared_ptr<Shape> s)
-{
-    if(s)
+auto perimeterBiggerThan20 = [](shared_ptr<Shape> s) {
+    if (s) {
         return (s->getPerimeter() > 20);
+    }
     return false;
-}
+};
 
-bool areaLessThan10(shared_ptr<Shape> s)
-{
-    if(s)
+auto areaLessThanX = [x = 10](shared_ptr<Shape> s) {
+    if (s) {
         return (s->getArea() < 10);
+    }
     return false;
-}
+};
 
-void printCollection(const Collection& collection)
-{
-    for (const auto & it : collection)
+void printCollection(const Collection& collection) {
+    for (const auto& it : collection)
         if (it)
             it->print();
```

Comment:
```suggestion
    for (const auto& it : collection)
        if (it)
            it->print();
```
```suggestion
    for (const auto& it : collection) {
        if (it) {
            it->print();
        }
    }
```

Reply:
I did not change this part of code (it was not a part of homework). If it is a very big problem you can make a PR to coders repo and maybe @ziobron will implement it. I know that refactoring old code is very useful, but I think I did a lot of this kind of refactor in the rest of files. I think that this comment will not improve my skills.

Reply:
@WBigaj as @Kasperice mentioned formatting is a personal matter. Your suggestion to change formatting on your preferred style makes code review harder not only because of your comments but also when people apply them I can see their changes which have nothing in common with homework. It makes me stop on every change and check if it is meaningful or simply ignore some parts of review and maybe sometimes overlook something.

Reply:
@WBigaj you can comment on inconsistent formatting but as far as it is consistent everything is ok and totally acceptable.